llvm/lib/Transforms/Vectorize/LoopVectorize.cpp

```diff
@@ -8318,6 +8323,15 @@ void LoopVectorizationPlanner::buildVPlansWithVPRecipes(ElementCount MinVF,
       if (CM.foldTailWithEVL())
         VPlanTransforms::runPass(VPlanTransforms::addExplicitVectorLength,
                                  *Plan, CM.getMaxSafeElements());
+
+      // See if we can convert an early exit vplan to bail out to a scalar
+      // loop if state-changing operations (like stores) are present and
```

Comment:
It looks like we currently also try to perform the conversion for non-state-changing loops?